mod/embed/views/default/elgg/embed.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * Embed module bootstraps the embed funtionality
+ * Note that this module will initialize itself once elgg/ready module is loaded,
+ * hence you do not need to call elgg/embed#init after requiring elgg/embed module.
+ *
+ * @module elgg/embed
+ */
+define(function (require) {
+
+	var elgg = require('elgg');
+	var $ = require('jquery');
+	var lightbox = require('elgg/lightbox');
+	require('jquery.form');
+
+	elgg.provide('elgg.embed');
+
+	var embed = {
+		/**
+		 * Flag used to avoid duplicate event bindings
+		 */
+		ready: false,
+		/**
+		 * Initializes the module
+		 * @return void
+		 */
+		init: function () {
+			if (embed.ready) {
+				return;
+			}
+			// inserts the embed content into the textarea
+			$(document).on('click', ".embed-item", embed.insert);
+			if (typeof elgg.embed._deprecated_custom_insert_js === 'function') {
+				elgg.register_hook_handler('embed', 'editor', elgg.embed._deprecated_custom_insert_js);
+			}
+			// caches the current textarea id
+			$(document).on('click', ".embed-control", function () {
+				var textAreaId = /embed-control-(\S)+/.exec($(this).attr('class'))[0];
+				embed.textAreaId = textAreaId.substr("embed-control-".length);
+			});
+			// special pagination helper for lightbox
+			$(document).on('click', '.embed-wrapper .elgg-pagination a', embed.forward);
+			$(document).on('click', '.embed-section', embed.forward);
+			$(document).on('submit', '.elgg-form-embed', embed.submit);
+			embed.ready = true;
```

Comment:
Move this right after the check for it. Assigning event handlers shouldn't take long, but why risk it...